src/step_definitions/sqs-steps.ts

```diff
@@ -35,6 +34,25 @@ export const sqsSteps: StepDefinitions = ({ given, then }) => {
     }
   );
 
+  then(
+    /queue with variable name "(.*)" should have message with the following MessageBody and MessageAttributes:/,
+    async (queueNameVariable: string, expectedMessage: string) => {
+      const message = await queue.getMessageInQueue(
+        process.env[queueNameVariable]
+      );
+
+      const parsedMessage = JSON.parse(expectedMessage);
```

Comment:
I am not sure if we should parse the message here, as the message body is actually expected to be a string (yes, I know in the scenario `queue with variable name "(.*)" should have message with following message Body:`, there is the same problem)
Parsing the body can make the json easier to read though, what do you think if we change the description to something like `following MessageBody\(JSON parsed\) and MessageAttributes:`

Reply:
Ahh, I see, sorry that I misread it, yes, Gherkin only support String for the parameter

Reply:
But then the tested scenario is strange
<img width="859" alt="image" src="https://github.com/shanwong29/gurke/assets/52825514/ae7dbb6e-55fb-4939-84eb-88df94468705">
In the given step, the messageBody is a stringified Json
But in the then step, the messageBody become a JSON object?? 🤔